### PR TITLE
Remove deprecated app module aliases.

### LINKED ISSUE #15618: Deprecated app module aliases should not cause warnings when building developer documentation.
### Is your feature request related to a problem? Please describe.
PR #13366 changed a way in which NVDA maps an app module to a given executable. In the process several old app modules were marked as deprecated (they show warnings on import). This is pretty noticeable when building developer documentation `scons devDocs`. The following two types of warnings are shown:
```
Importing appModules.digitaleditionspreview is deprecated, instead import appModules.digitaleditions.

```

```
WARNING: invalid signature for automodule ('appModules.azardi-2_0')
WARNING: don't know which module to import for autodocumenting 'appModules.azardi-2_0' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
WARNING: invalid signature for automodule ('appModules.azuredatastudio-insiders')
WARNING: don't know which module to import for autodocumenting 'appModules.azuredatastudio-insiders' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
WARNING: invalid signature for automodule ('appModules.code - insiders')
WARNING: don't know which module to import for autodocumenting 'appModules.code - insiders' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)

```

### Describe the solution you'd like
Since these are deprecated aliases (with the exception of module for Azardi, see below) I suggest to remove them in the 2024.1 development cycle. We cannot just disable deprecated code when building developer documentation, since for most of these modules they cannot be imported by Sphinx (their names do not comply to the naming requirement of a Python module). Module for Azardi is a special case - it is not deprecated. However since we are in the backward compatibility breaking release, I suggest to just rename it into something like `azardi20`, and map it to the Azardi's binary.
### Describe alternatives you've considered
If it is decided that keeping backward compat here is desired, it should be possible to write a custom importer which simulates presence of these modules at runtime. I'm willing to work on this, if that is the direction NV Access prefers.
### Additional context
While I believe keeping backward compatibility is important, the deprecated code should not be getting in the way. I'm also not aware of any add-on which relies on presence of these deprecated app modules.

### PR DESCRIPTION
**Pull request by lukaszgo1**
### Link to issue number:
Closes #15618

### Summary of the issue:
PR #13366 changed a way in which NVDA maps an app module to a given executable. In the process several old app modules were marked as deprecated (they show warnings on import). This is pretty noticeable when building developer documentation `scons devDocs`.
### Description of user facing changes
None
### Description of development approach
Deprecated app modules and supporting functions from `appModuleHandler` were removed. The app Module for Azardi  which cannot be imported by Sphinx was renamed, and mapped to Azardi's binary.
### Testing strategy:
Ensured that app module for Azardi is loaded. Ensured that warnings when building developer documentation are gone.
### Known issues with pull request:
None known
### Code Review Checklist:

- [X] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [X] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [X] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [X] API is compatible with existing add-ons.
- [X] Security precautions taken.

### CODE REVIEW

**Comment by CyrilleB79** [comment]
In the future, if there are appModules to rename, does it mean that it will not be allowed anymore to write deprecation code to map the old name to the new appModule? Or otherwise we would reintroduce #15618?

**Comment by lukaszgo1** [comment]
> In the future, if there are appModules to rename, does it mean that it will not be allowed anymore to write deprecation code to map the old name to the new appModule? Or otherwise we would reintroduce #15618?



If name of old app module would be valid (i.e. would not contain characters such as `+` or `.`), then it should be possible to show deprecation warning on import only when not building developer documentation. DO you have a particular rename in mint?

**Comment by lukaszgo1** [comment]
As I mentioned in #12971 we probably should enable building developer documentation in CI, and fail the build on any newly introduced warnings.

File: user_docs/en/changes.t2t [hunk 1]
```diff
@@ -73,8 +73,38 @@ That method receives a ``DriverRegistrar`` object on which the ``addUsbDevices``
 - ``IoThread.autoDeleteApcReference`` has been removed. (#14924, @LeonarddeR)
 - To support capital pitch changes, synthesizers must now explicitly declare their support for the ``PitchCommand`` in the ``supportedCommands`` attribute on the driver. (#15433, @LeonarddeR)
 - ``speechDictHandler.speechDictVars`` has been removed. Use ``NVDAState.WritePaths.speechDictsDir`` instead of ``speechDictHandler.speechDictVars.speechDictsPath``. (#15614, @lukaszgo1)
+% Insert new list items here as the alias appModule table should be kept at the bottom of this list
+- The following app modules are removed.
+Code which imports from one of them, should instead import from the replacement module.  (#15618, @lukaszgo1)
 -
 
+|| Removed module name | Replacement module |
+| ``azardi-2.0`` | ``azardi20``
```
Comment (seanbudd) [inline user_docs/en/changes.t2t:82]:
```suggestion
| ``azardi-2.0`` | ``azardi20`` |
```

**Review by seanbudd — APPROVED** [review]
Thanks @lukaszgo1

**Pull request merged**